src/cmd/prune_branches.go

```diff
@@ -38,6 +41,12 @@ This usually means the branch was shipped or killed on another machine.`,
 }
 
 func checkPruneBranchesPreconditions() {
+	if git.IsOffline() {
+		fmt.Println(`Error: cannot prune branches in offline mode
+
+Pruning branches requires an active internet connection.`)
```

Comment:
I have added `util.ExitBecauseOffline`. Please note that this PR also updates `git new-pull-request`. The remaining commands are updated in their respective PRs.